prepare_data.py

```diff
@@ -38,6 +87,8 @@ def main():
                     help="path to save prepared data")
     ap.add_argument("--val_size", type=float, default=0.2,
                     help="validation split ration")
+    ap.add_argument("--cores", type=int, help='Number of virtual cores to parallelize over',
+                    default=max(1, (mp.cpu_count() - 2) / 2)) # To prevent IPC issues
```

Comment:
It defaults to a float value if we don't use floor division

```suggestion
                    default=max(1, (mp.cpu_count() - 2) // 2)) # To prevent IPC issues
```